server/utils/forms/inputs/calendarDayInput.ts

```diff
@@ -95,8 +96,52 @@ export default class CalendarDayInput {
         .withMessage(this.errorMessages.yearEmpty)
         .bail()
         .trim()
-        .isInt({ max: 9999 }) // sure, dates bigger than this are real dates, but they're not valid here
+        .isInt({
+          min: 2000,
+          max: 9999,
+        }) // sure, dates bigger than this are real dates, but they're not valid here - minimum date protects from submission errors
         .withMessage(this.errorMessages.invalidDate),
+      ExpressValidator.body(this.keys.day).custom(() => {
+        const enteredDate = new Date(
+          Number(this.request.body[this.keys.year]),
+          Number(this.request.body[this.keys.month] - 1),
+          Number(this.request.body[this.keys.day]),
+          23,
+          59,
+          59
+        )
+
+        if (this.referralDate != null) {
+          const maximumDate = new Date()
+          maximumDate.setMonth(maximumDate.getMonth() + 6)
```

Comment:
@JakeGoodwin22 Thanks for these!

As this is a business rule, can we also implement this in the domain API, please? It should reject bookings with appointment times not meeting this criterion.